### ISSUE TITLE
Support Country > State drills

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#36659]
This should be part of the zoom-in geo drill. This is a nice thing to have, not a blocking bug.

There was code for this on the frontend, but after a lot of head scratching we could not figure out how to actually make it trigger and as far as we knew no one had ever actually seen it working... we made the decision not to port it at the time we ported the rest of the zoom-in geo stuff. See https://github.com/metabase/metabase/issues/33557#issuecomment-1745850219 and https://metaboat.slack.com/archives/C04CYTEL9N2/p1696373468465189?thread_ts=1696373304.645749&cid=C04CYTEL9N2 . 

It sounds like @ranquild has actually found a very specific scenario where you are able to trigger it on the FE with MLv1, see https://metaboat.slack.com/archives/C04CYTEL9N2/p1701826072570649. 

Either way, we should get it working in MLv2. We should unskip these FE unit tests

https://github.com/metabase/metabase/blob/e87c11e923a39e1a50a5119440354b3ae0dbce05/frontend/src/metabase-lib/tests/drills-zoom-in-geographic.unit.spec.ts#L64-L87

and write Clojure tests as well.

**Comment by paoliniluis** [comment]
Can we make this parent->child (like state-> post code or state->city)?

**Comment by camsaul** [comment]
yes, it's like the existing state -> city. We don't support state -> post code.
